### DISCUSSION TITLE
OrcaSlicer 2.1 and Moonraker Connect issue

### DISCUSSION THREAD

**Discussion by Buford7** [OrcaSlicer/OrcaSlicer#7306] [Q&A]
Hello--I have a rooted K1C printer with Fluidd and Orcaslicer.  All are updated along with Moonraker to current stable releases.  I am having a problem with the Device tab in Orcaslicer where it constantly requires a "Force" refresh to get Moonraker started.  Is this a known bug, or any way to get Moonraker to automatically start vs having to do a Force Refresh every time I click into the Device tab of Orca?  (Moonraker does work once Force Refresh is clicked).  I have attempted to attach my screenshot of the issue.  Thanks for any ideas!

![FluiddMoonrakerOrcaSlicer 2 1](https://github.com/user-attachments/assets/baf82cd0-87e7-42dc-8913-09d75cb084e9)

**Comment by basilverthorn** [comment]
Is you port correct? It is usually 4408 for fluid, works near perfect here.

4409 - default port of Mainsail on the K1
4408 - default port of Fluidd on the K1

Guessing you manaualy set the port?

**Comment by Buford7** [comment]
Hello… Yes, I do have my port set to 4408. It just doesn’t connect automatically and I have to use the force refresh button within the Orca device tab to get the connection working within orca. And yes, I did have to put the port in manually within orca to get any of that going. 

Since it does ultimately work, just not automatically as expected- I imagine there’s nothing I need to do with my router? Regardless, thanks for your thoughts, and if you have any other suggestions, I’m certainly willing to try anything.

**Comment by Buford7** [comment]
But as I look at my screenshot above I’m reminded that the Fluidd port that you put into Orca setup gets forwarded somehow to Moonraker (which is the 7125 that shows up after my IP address). I’ve tried setting moonraker as the port in Orca device setup and that doesn’t work at all. Seems that  Fluidd has to be the first in line which then connects to Moonraker to produce the visual interface in the Orca device tab.  That’s how I understand it at least.

**Comment by basilverthorn** [comment]
You are welcome, none other than the screen shot above on that port. Very odd, nothing in your router is set to do any port forwarding?
My entry in Orca is as follows: http://192.168.1.136:4408, and in checking the device tab it does not change to the 71XX port.

![image](https://github.com/user-attachments/assets/b84ef9b8-4377-4426-8a6a-cd55d3e8c1ef)

**Comment by basilverthorn** [comment]
Correct if you enter that IP in your browser on port 7125 you should get moonraker API welcome page. Just odd as Fluidd, usually does this in the background. I have not seen it on my installs of Orca.

**Reply by Buford7** [reply]
Alright—the issue was that I did not have the HTTP designation in front of the IP address! I corrected that and put in everything in the same format that you did and now it connects to fluid directly without any issue with an intermediate requirement to do a force connection!

Thank you so much for talking this out with me and I seem to be off to the races now. Much appreciated!

**Comment by basilverthorn** [comment]
Hey AWESOME, glad to hear I could help. Happy Printing!!!

**Reply by Buford7** [reply]
Community supports works well for sure👍

**Reply by Buford7** [reply]
Well shoot. Just went back to orca after having laptop closed and I’m back to having the same screen as my original screenshot with it requiring Force Restart vs just going right into Fluidd.  So I guess I’m back to square one.  Perhaps there is something with Mac version of Orca, or something with my router that needs to be changed.  May be beyond me at this point.  Any suggestions if I should be posting this elsewhere in the Orca GitHub?

**Comment by basilverthorn** [comment]
I’ll try it on my Mac tomorrow.  Is 2.2 out for Mac? I’ll look later tonight or tomorrow.  Sent from my iPhoneOn Nov 6, 2024, at 1:48 PM, Buford7 ***@***.***> wrote:﻿
Well shoot. Just went back to orca after having laptop closed and I’m back to having the same screen as my original screenshot with it requiring Force Restart vs just going right into Fluidd.  So I guess I’m back to square one.  Perhaps there is something with Mac version of Orca, or something with my router that needs to be changed.  May be beyond me at this point.  Any suggestions if I should be posting this elsewhere in the Orca GitHub?

**Comment by Buford7** [comment]
Thanks for your help!  I did try adding Mainsail as an alternate option to Fluidd and it certainly doesn't have the same issue.  The interface is pretty much similar to Fluidd, so I will try working with that for a bit and see how it goes.  

You are right that  2.2 Orca is now available-there is no mention of this specific Fluidd/Moonraker connect issue in their bug fixes, but will be worth a try as well.  I have downloaded it and set up with Fluidd and will see how well it holds the connection.

**Reply by Buford7** [reply]
Looks like Fluidd still requires the Force Reload on Orca 2.2 after Orca being open for awhile, Mainsail either holds the connection or reloads itself.  (I have been leaving Orca open on my Mac M3 Air laptop but letting screensaver go on and checking back on the program after 10-30min, so simulating to some extent what I would be doing while a print is in progress).

**Comment by KaiserLeo7** [comment]
This is probably the dumbest issue I've ran into trying out Cura, Prusa, and lastly Orca.

I go to connect my Neptune 4 Plus and I enter my IP with Host Type as Octo/Klipper (ChatGPT is telling me i have the wrong fork/version as the last one had Moonraker option)

I enter my IP press Browse, press Test. I get and Error no such port blah blah blah

I mentally made note that until the printer is connected the Device TAB only shows a short video on how to connect the Printer.

So figure me surprised that after entering my IP and still getting the error I switch to the Device TAB and there it is... Fluidd connected to my printer.

Well done guys.
